internal/pool/pool.go

```diff
@@ -249,22 +251,23 @@ func (p *ConnPool) getTurn() {
 }
 
 func (p *ConnPool) waitTurn(ctx context.Context) error {
-	var done <-chan struct{}
-	if ctx != nil {
-		done = ctx.Done()
+	if ctx == nil {
+		// The background context's Done is nil by default.
+		ctx = context.Background()
 	}
 
 	select {
-	case <-done:
+	case <-ctx.Done():
 		return ctx.Err()
```

Comment:
The old code has a bug on this line (259).  This version fixes that bug.

Reply:
Sorry I should clarify this wasn't a bug, but instead a static analysis issue.  You block on 'done', but return ctx.Err, but allow ctx to be nil.  It fails linting because it's very strange to return a parameter value of something you've previously checked to allow nil, but in the current setup of the code it's fine since done is nil.  I've never seen `if ctx != nil` checks in Go code.  Usually these become context.TODO or context.Background as soon up the call stack as possible, like I'm doing here.

I can revert if you like

Reply:
I agree that it is better to use context.TODO or context.Background instead of nil, but it should happen higher in the hierarchy - not here.